pkg/hardware/config.go

```diff
@@ -69,7 +69,7 @@ func (hc *HardwareConfig) setHardwareConfigFromFile(hardwareFileName string) err
 	return nil
 }
 
-func (hc *HardwareConfig) ValidateHardware() error {
+func (hc *HardwareConfig) ValidateHardware(skipBmcCheck bool) error {
 	bmcRefMap := hc.initBmcRefMap()
```

Comment:
It would be nice to not call this initBmcRefMap in the skipBmcCheck=true case since it will not get populated/used

Reply:
Going to lgtm this PR for now, since all looks good. You can address this minor comment in a follow-up later.